src/main/java/com/kryptnostic/api/v1/storage/DefaultStorageClient.java

```diff
@@ -151,17 +146,17 @@ private void makeObjectSearchable( KryptnosticObject object, EncryptedSearchShar
 
     }
 
-    private EncryptedSearchSharingKey setupSharing( KryptnosticObject object ) throws IrisException {
+    private IndexPair setupSharing( KryptnosticObject object ) throws IrisException {
         Stopwatch watch = Stopwatch.createStarted();
-        // generate nonce
-        EncryptedSearchSharingKey sharingKey = context.generateSharingKey();
+        KryptnosticEngine engine = context.getConnection().getKryptnosticEngine();
+        IndexPair indexPair = IndexPair.newFromKryptnosticEngine( engine );
+        byte[] sharingPair = indexPair.computeSharingPair( engine );
```

Comment:
yea, I don't see a reason why `computeSharingPair()` should be a method on `IndexPair`, especially having seen the implementation. it should be a method on `KryptnosticEngine`